### PR TITLE
Consolidate cloud build between generic and autopilot

### PR DESCRIPTION
**Pull request by chiayi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:
There isn't a difference between generic and autopilot, so we clean up the cloud build yaml conditional.

**Which issue(s) this PR fixes**:
Part of https://github.com/googleforgames/agones/pull/3165#discussion_r1199563868

**Special notes for your reviewer**:

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -312,18 +312,10 @@ steps:
         for version in "${!versionsAndRegions[@]}"
         do
           region=${versionsAndRegions[$version]}
-          if [ $cloudProduct = generic ]
-          then
-            featureWithGate="CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=false&PodHostname=false&SplitControllerAndExtensions=false&FleetAllocationOverflow=true&Example=true"
-            featureWithoutGate=""
-            testClusterLocation="${region}-c"
```
Comment (gongmax) [inline cloudbuild.yaml:319]: We cannot do this before move the Standard cluster to regional.

Comment (chiayi) [reply cloudbuild.yaml:319]: Ooooooh I see! Okay I completely missed that. Thanks!

**Comment by chiayi** [comment]
This will be in draft until the Standard clusters will be in regional. If for some reason, the standard clusters will never be in regional, this pr can then be deleted.

File: cloudbuild.yaml (hunk 1 above)
Comment (roberthbailey) [reply cloudbuild.yaml:319]: I think it's just the `featureWithGate` and `featureWithoutGate` variables that get collapsed.

Comment (chiayi) [reply cloudbuild.yaml:319]: I see, so that means there isn't a need for a `featureWithoutGate` anymore. Just one variable `featurewithGate`

**Review by gongmax — APPROVED** [review]

**Review by gongmax — DISMISSED** [review]

File: cloudbuild.yaml (hunk 1 above)
Comment (roberthbailey) [reply cloudbuild.yaml:319]: Correct. You should be able to clean up the usages of `featureWithoutGate` and (unless I'm misreading it) collapse the two places where `featurewithGate` is set into one.

File: cloudbuild.yaml [hunk 2]
```diff
@@ -323,7 +323,7 @@ steps:
           featureWithoutGate=""
           testClusterLocation="${region}"
           { stdbuf -oL -eL gcloud builds submit . --suppress-logs --config=./ci/e2e-test-cloudbuild.yaml \
-            --substitutions _FEATURE_WITH_GATE=$featureWithGate,_FEATURE_WITHOUT_GATE=$featureWithoutGate,_CLOUD_PRODUCT=$cloudProduct,_TEST_CLUSTER_NAME=$testCluster,_TEST_CLUSTER_LOCATION=$testClusterLocation,_REGISTRY=${_REGISTRY},_PARENT_COMMIT_SHA=${COMMIT_SHA},_PARENT_BUILD_ID=${BUILD_ID} \
+            --substitutions _FEATURE_WITH_GATE=$featureWithGate,_CLOUD_PRODUCT=$cloudProduct,_TEST_CLUSTER_NAME=$testCluster,_TEST_CLUSTER_LOCATION=$testClusterLocation,_REGISTRY=${_REGISTRY},_PARENT_COMMIT_SHA=${COMMIT_SHA},_PARENT_BUILD_ID=${BUILD_ID} \
```
Comment (roberthbailey) [inline cloudbuild.yaml:326]: If it are no longer used, can you remove the `featureWithoutGate` variable on lines 323 above?

Comment (roberthbailey) [inline cloudbuild.yaml:326]: Unless I'm misreading them, the two places we set `featureWithGate` are now the same and can be consolidated into a single variable (e.g. moved outside of the if statement). 

The only thing that would remain inside of the `if [ $cloudProduct = generic ]` block is how we pick the name of the test cluster.

Comment (gongmax) [reply cloudbuild.yaml:326]: I think `featureWithGate` are still different. For Generic, `SplitControllerAndExtensions` is set to false while in Autopilot it's set to true

Comment (chiayi) [reply cloudbuild.yaml:326]: I think we can set both to be true now that its in beta?

Comment (gongmax) [reply cloudbuild.yaml:325]: oh that makes sense. We can set both to true.

Comment (roberthbailey) [reply cloudbuild.yaml:325]: Looking at https://github.com/googleforgames/agones/pull/3165 again, I think we may need to wait until we hit stable to consolidate the variables. That PR flipped the `featureWithGate` for non-autopilot to false whereas on autopilot it needs to stay true.

**Comment by roberthbailey** [comment]
Thanks for confirming. I'm going to close this PR and we will consolidate once the feature graduates to stable.

**Pull request closed without merging**